code/datums/status_effects/buffs.dm

```diff
@@ -188,15 +191,52 @@
 		return
 	REMOVE_TRAIT(owner, TRAIT_CHUNKYFINGERS, VAMPIRE_TRAIT)
 	var/mob/living/carbon/human/H = owner
-	H.physiology.brute_mod /= 0.5
-	H.physiology.burn_mod /= 0.8
+	H.color = original_color
+	H.physiology.brute_mod /= 0.4
+	H.physiology.burn_mod /= 0.5
 	H.physiology.stamina_mod /= 0.5
 	H.physiology.stun_mod /= 0.5
 	if(bonus_damage_applied)
 		bonus_damage_applied = FALSE
 		H.physiology.melee_bonus -= 10
 		H.dna.species.punchstunthreshold -= 8
 
+/datum/status_effect/vampire_gladiator
+	id = "vampire_gladiator"
+	duration = 30 SECONDS
+	tick_interval = 1 SECONDS
+	alert_type = /obj/screen/alert/status_effect/vampire_gladiator
+	var/healing_ticks_left // We use this to see how long we can keep healing the vampire
+
+/obj/screen/alert/status_effect/vampire_gladiator
+	name = "Gladiatorial resilience"
+	desc = "Roused by the thrill of the fight, your body has become more resistant to breaking!"
+	icon = 'icons/mob/actions/actions.dmi'
+	icon_state = "mech_damtype_brute"
+
+/datum/status_effect/vampire_gladiator/on_apply()
+	. = ..()
+	healing_ticks_left = 30
+	var/mob/living/carbon/human/H = owner
+	for(var/Y in H.bodyparts)
+		var/obj/item/organ/external/E = Y
+		E.min_broken_damage += 100 // Arbitrary high damage number, we just want to stop bone breaks from happening
```

Comment:
dont do this. 
instead, try and make use of this proc
```
/obj/item/organ/external/proc/add_limb_flags()
	if(HAS_TRAIT(owner, TRAIT_NO_BONES))
		limb_flags |= CANNOT_BREAK
		encased = null // no bones to saw

	if(HAS_TRAIT(owner, TRAIT_STURDY_LIMBS))
		limb_flags |= CANNOT_DISMEMBER

	if(HAS_TRAIT(owner, TRAIT_BURN_WOUND_IMMUNE))
		limb_flags |= CANNOT_BURN
```

you will need to make a `remove_limb_flags` proc tho that does the opposite.

Reply:
Alright so, I've slightly deviated of using the proc, and instead put all the checks of the proc inside the code. The reason being `encased` being set to `null`, which if we'd want to restore that later, we'd have to check if the limb was the chest or skull, for every limb.

I've also made a new flag for preventing IB, as that was not covered in the flags in `add_limb_flags()`

Reply:
Why not just set encased to initial()

Reply:
Because I really just forgot that was a thing, I'll go update the code in a bit